src/logics/sonne/logic.supply.test.ts

```diff
@@ -0,0 +1,80 @@
+import { CErc20Immutable__factory } from './contracts';
+import { LogicTestCase } from 'test/types';
+import { SupplyLogic, SupplyLogicFields } from './logic.supply';
+import * as common from '@protocolink/common';
+import { constants, utils } from 'ethers';
+import * as core from '@protocolink/core';
+import { expect } from 'chai';
+import { optimismTokens } from './tokens';
+import { toCToken } from './configs';
+
+describe('Sonne SupplyLogic', function () {
+  context('Test getTokenList', async function () {
+    SupplyLogic.supportedChainIds.forEach((chainId) => {
+      it(`network: ${common.toNetworkId(chainId)}`, async function () {
+        const logic = new SupplyLogic(chainId);
+        const tokenList = await logic.getTokenList();
+        expect(tokenList).to.have.lengthOf.above(0);
+      });
+    });
+  });
+
+  context('Test build', function () {
+    const chainId = common.ChainId.optimism;
+    const logic = new SupplyLogic(chainId);
+    const ifaceCErc20 = CErc20Immutable__factory.createInterface();
+
+    const testCases: LogicTestCase<SupplyLogicFields>[] = [
+      {
+        fields: {
+          input: new common.TokenAmount(optimismTokens.ETH, '1'),
+          output: new common.TokenAmount(toCToken(chainId, optimismTokens.ETH.wrapped), '0'),
```

Comment:
I prefer to pass the wrapped address to `toCToken` because `toCToken` is to find the direct mapping in the Sonne config.

The supply test is to test the supply token behavior. We passed `ETH.wrapped` to `toCToken` to mimic the Protocolink wrap token behavior which seems pretty legit.

Reply:
Sounds good.